kratos/solving_strategies/schemes/residual_based_bossak_displacement_scheme.hpp

```diff
@@ -239,75 +244,85 @@ class ResidualBasedBossakDisplacementScheme
      * @details It predicts the solution for the current step x = xold + vold * Dt
      * @param rModelPart The model of the problem to solve
      * @param rDofSet set of all primary variables
-     * @param A LHS matrix
-     * @param Dx Incremental update of primary variables
-     * @param b RHS Vector
+     * @param rA LHS matrix
+     * @param rDx Incremental update of primary variables
+     * @param rb RHS Vector
      */
     void Predict(
         ModelPart& rModelPart,
         DofsArrayType& rDofSet,
-        TSystemMatrixType& A,
-        TSystemVectorType& Dx,
-        TSystemVectorType& b
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb
         ) override
     {
         KRATOS_TRY;
 
-        const double delta_time = rModelPart.GetProcessInfo()[DELTA_TIME];
+        // The current process info
+        const ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
+        const double delta_time = r_current_process_info[DELTA_TIME];
 
         // Updating time derivatives (nodally for efficiency)
-        const int num_nodes = static_cast<int>(rModelPart.NumberOfNodes());
+        const int num_nodes = static_cast<int>( rModelPart.Nodes().size() );
+        const auto it_node_begin = rModelPart.Nodes().begin();
+
+        // Getting position
+        const int disppos = it_node_begin->HasDofFor(DISPLACEMENT_X) ? it_node_begin->GetDofPosition(DISPLACEMENT_X) : -1;
+        const int velpos = it_node_begin->HasDofFor(VELOCITY_X) ? it_node_begin->GetDofPosition(VELOCITY_X) : -1;
+        const int accelpos = it_node_begin->HasDofFor(ACCELERATION_X) ? it_node_begin->GetDofPosition(ACCELERATION_X) : -1;
+
+        // Getting dimension
+        const std::size_t dimension = r_current_process_info.Has(DOMAIN_SIZE) ? r_current_process_info.GetValue(DOMAIN_SIZE) : 3;
```

Comment:
I would throw an error if the DOMAIN_SIZE is not there

Reply:
Several tests fail, let it be this way, at least for now

Reply:
cannot you update the tests beforehand? It shouldn't hurt adding DOMAIN_SIZE there, then you can do it properly here

Reply:
alternatively let's just detect it a different way

If(nodes_being->HasDOF(DISPLACEMENT_Z) || nodes_being->HasDOF(VELOCITY_Z) || nodes_being->HasDOF(VELOCITY_Z) domain_size = 3
elseif ... check for y ... domain_size = 2
else domain_size = 1

this would be consistent with the use you do later